trieste/space.py

```diff
@@ -65,17 +67,10 @@ def __pow__(self: SP, other: int) -> SP:
         :param other: The exponent, or number of instances of this search space to multiply
             together. Must be strictly positive.
         :return: The Cartesian product of ``other`` instances of this search space.
-        :raise ValueError: If the exponent ``other`` is less than 1.
+        :raise tf.errors.InvalidArgumentError: If the exponent ``other`` is less than 1.
         """
-        if other < 1:
-            raise ValueError(f"Exponent must be strictly positive, got {other}")
-
-        space = self
-
-        for _ in range(other - 1):
-            space *= self
-
-        return space
+        tf.debugging.assert_positive(other, message="Exponent must be strictly positive")
+        return reduce(operator.mul, [self] * other)
```

Comment:
felt like doing this line, cos it's neat

Reply:
definitely more elegant! 
am I missing something or `assert_positive` means >0 while here we want >=1?

Reply:
you're right. maybe i though positive meant strictly positive

Reply:
scrap that ... > 0 is the same as >=1 for integers, and that's also what `assert_positive` does

Reply:
> `tf.debugging` works as expected with `tf.function`. `raise` doesn't (it will only raise when the function is compiled). `assert` is only used in source code for internal correctness checks. I don't believe trieste uses `assert` outside tests (`assert` is fine in tests, if you're not using `tf.function`)

I see, I didn't know about the difference with respect to compilation. 
So `assert` has the same issue if tf.function is used? 
Actually, I have noticed we are dominantly using if-else accompanied with raising an error, while to me it usually seems like it should be an assert statement?

Regarding `tf.function`, at the moment it is used only in model optimization, is that correct?